core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -135,6 +140,40 @@ public String serialize() {
     return sb.toString();
   }
 
+  /**
+   * Returns true if the serialized fingerprint matches the fingerprint in metadata.
+   *
+   * @param serializedFP a finger print in string representation
+   *
+   * @return true if the serialized fingerprint matches the current fingerprint in metadata
+   */
+  public boolean matchMetadata(String serializedFP) {
+    Fingerprint fp = Fingerprint.parse(serializedFP);
+    for (Tag tag : METADATA_TAGS) {
+      if (!this.getTag(tag).equals(fp.getTag(tag))) {
```

Comment:
we follow of a style of not using `this.` unless necessary - I think it can be omitted here and in `matchContent`